src/main/scala/strawman/collection/View.scala

```diff
@@ -86,6 +86,20 @@ object View extends IterableFactory[View] {
     def iterator(): Iterator[A] = Iterator.iterate(start)(f).take(len)
     override def knownSize: Int = 0 max len
   }
+  
+  case class Intersperse[A, B >: A](underlying: Iterable[A], sep: B) extends View[B] {
```

Comment:
You should use the type parameter `A` everywhere instead of having both an `A` and a `B`.